autonomic-plugin-common/src/test/java/br/com/autonomiccs/autonomic/plugin/common/daos/HostJdbcDaoTest.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * This program is part of Autonomiccs "autonomic-platform",
+ * an open source autonomic cloud computing management platform.
+ * Copyright (C) 2016 Autonomiccs, Inc.
+ *
+ * Licensed to the Autonomiccs, Inc. under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The Autonomiccs, Inc. licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package br.com.autonomiccs.autonomic.plugin.common.daos;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+import com.cloud.host.Status;
+import com.cloud.hypervisor.Hypervisor.HypervisorType;
+import com.cloud.resource.ResourceState;
+
+import br.com.autonomiccs.autonomic.plugin.common.enums.HostAdministrationStatus;
+import br.com.autonomiccs.autonomic.plugin.common.enums.StartType;
+
+@RunWith(MockitoJUnitRunner.class)
+public class HostJdbcDaoTest {
+
+    private HostJdbcDao spy;
+    private JdbcTemplate jdbcTemplate;
+
+    private String sqlGetAdministrationStatus = "SELECT administration_status FROM host WHERE id=?;";
+    private String sqlSetAdministrationStatus = "UPDATE host SET administration_status=? WHERE id=?;";
+    private String sqlGetStartType = "SELECT start_type FROM host WHERE id=?;";
+    private String sqlGetStatus = "SELECT status FROM host WHERE id=?;";
+    private String sqlGetResourceState = "SELECT resource_state FROM host WHERE id=?;";
+    private String sqlSetAllHypervisorsTypeInCloud = "select hypervisor_type from host where removed is null and hypervisor_type is not null group by hypervisor_type";
+    private String sqlCheckIsThereAnyHostOnCloudDeactivatedByOurManager = "select id from host where removed is null and administration_status = 'ShutDownToConsolidate'";
+    private String sqlCheckIsThereAnyHostOnPodDeactivatedByOurManager = "select id from host where removed is null and administration_status = 'ShutDownToConsolidate' and pod_id = ?";
+
+    @Before
+    public void setup() {
+        spy = Mockito.spy(new HostJdbcDao());
```

Comment:
Hello @GabrielBrascher.
Had you some reason to create this method setup() instead of using the @spy and @mock annotations like you do in the anothers tests?
The @spy don't mock the properties of spied instances?

Thanks.